raiden/transfer/utils.py

```diff
@@ -53,3 +58,12 @@ def is_valid_secret_reveal(
     transfer_secrethash: SecretHash,
 ) -> bool:
     return state_change.secrethash == transfer_secrethash
+
+
+def encrypt_secret(secret: Secret, target_metadata: Optional[AddressMetadata]) -> EncryptedSecret:
+    encrypted_secret = EncryptedSecret(b"")
+    target_metadata = target_metadata or {}
+    public_key = target_metadata.get("displayname")
```

Comment:
If `target_metadata` is provided (i..e. not `None`), isn't `displayname` supposed to be there?

Reply:
I'm not so sure, but I don't think this value can be taken for granted. Maybe @fredo can jump in and clarify?

Reply:
@istankovic is right. The displayname is the signature of the `user_id` of the peer signed with the private key. This means that with the `user_id`, you can recover the public key from the signature.
We definitely can take the displayname for granted as it proofs that the holder of the private key with whom we want to interact signed the corresponding user_id. 
The check if the signature is correct and this is a vaild displayname should actually always happen whenever the node receives metadata. It authenticates the user_id. 

That's probably only partially checked yet and should happen whenever the raiden node consumes metadata (upon `query_metadata` and when receiving metadata for a route). We can outsource this into a separate issue. For @netcriptus PR we can simply assume that the data already have been checked against.

Reply:
PS: We should always drop metadata if the displayname is not there or does not authenticate the corresponding user_id.